### ISSUE TITLE
The penalty function and the loss function

### ISSUE THREAD

**Issue by kristinmg** [Statistical-Connectomics-Sp15/intro#114]
I know the penalty is usually a function of the number of samples and parameters in the model, i.e. we are penalizing for complexity.. But is/can the penalty also be related to the loss function?

**Comment by mrjiaruiwang** [comment]
I was under the impression that penalty is a term you can add to the loss function. For example, if you want to penalize for the number of parameters used in the model, the penalty term you could add would be the size of parameters raised to some power to control how much you want to penalize. A good example of this would be to penalize the size of weights in a perceptron learning scheme. This is done in an attempt to decrease variance in terms of the bias-variance tradeoff. I thought this was similar to the max-margin method in SVMs

**Comment by mblohr** [comment]
From http://stackoverflow.com/questions/25042909/difference-between-penalty-and-loss-parameters-in-sklearn-linearsvc-library : 
"In machine learning, loss function measures the quality of your solution, while penalty function imposes some constraints on your solution."
